src/main/java/uk/gov/hmcts/reform/unspec/service/BusinessProcessService.java

```diff
@@ -0,0 +1,38 @@
+package uk.gov.hmcts.reform.unspec.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.unspec.model.BusinessProcess;
+import uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.DISPATCHED;
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.READY;
+import static uk.gov.hmcts.reform.unspec.model.BusinessProcessStatus.STARTED;
+
+@Service
+@RequiredArgsConstructor
+public class BusinessProcessService {
+
+    private static final List<BusinessProcessStatus> ERROR_STATUSES = List.of(READY, DISPATCHED, STARTED);
+    private static final String ERROR_MESSAGE = "Business Process Error";
+
+    private final ObjectMapper mapper;
+
+    public List<String> updateBusinessProcess(Map<String, Object> data, String businessProcessEvent) {
+        BusinessProcess businessProcess = mapper.convertValue(data.get("businessProcess"), BusinessProcess.class);
+        if (Optional.ofNullable(businessProcess)
```

Comment:
I would actually be a even more defensive. Could we check that the business process `STATUS` is either null or `FINISHED` to be permitting a reset to `READY`. The logic is the same as here:
https://github.com/hmcts/unspec-service/pull/242/files#diff-4c2ee2ed8f256169482f5b867f57ea2dR53

Maybe extracting this in a utility class to be shared?